src/algebra/associated.lean

```diff
@@ -547,6 +550,56 @@ begin
   rw [mk_mul_mk, mk_dvd_mk, mk_dvd_mk, mk_dvd_mk],
 end
 
+theorem irreducible_mk_iff (a : α) : irreducible (associates.mk a) ↔ irreducible a :=
+begin
+  simp [irreducible, is_unit_mk],
+  apply and_congr iff.rfl,
+  split,
+  { assume h x y eq,
+    have : is_unit (associates.mk x) ∨ is_unit (associates.mk y),
+      from h _ _ (by rw [eq]; refl),
+    simpa [is_unit_mk] },
+  { refine assume h x y, quotient.induction_on₂ x y (assume x y eq, _),
+    rcases quotient.exact eq.symm with ⟨u, eq⟩,
+    have : a = x * (y * u), by rwa [mul_assoc, eq_comm] at eq,
+    show is_unit (associates.mk x) ∨ is_unit (associates.mk y),
+    simpa [is_unit_mk] using h _ _ this }
+end
```

Comment:
Basically two changes: `simp` that is not at the end of a proof should be `simp only` for robustness, and renamed the hypothesis `eq` with `a_eq`, so that it's clearer that `eq.symm` is actually `_root_.eq.symm eq`.
```suggestion
begin
  simp only [irreducible, is_unit_mk],
  apply and_congr iff.rfl,
  split,
  { rintro h x y rfl,
    simpa [is_unit_mk] using h (associates.mk x) (associates.mk y) rfl },
  { intros h x y,
    refine quotient.induction_on₂ x y (assume x y a_eq, _),
    rcases quotient.exact a_eq.symm with ⟨u, a_eq⟩,
    rw mul_assoc at a_eq,
    show is_unit (associates.mk x) ∨ is_unit (associates.mk y),
    simpa [is_unit_mk] using h _ _ a_eq.symm }
end
```